lib/spack/spack/build_environment.py

```diff
@@ -161,6 +161,23 @@ def clean_environment():
     env.unset('LD_RUN_PATH')
     env.unset('DYLD_LIBRARY_PATH')
 
+    # Remove language-related vars, so compiler messages etc. are not
```

Comment:
@michaelkuhn a couple things:

* I'm wondering if we should explicitly set the language variables vs. unsetting them. Based on https://unix.stackexchange.com/questions/87745/what-does-lc-all-c-do, setting `LC_ALL` would override these.
* Ideally the error parsing would handle other languages, but the regexes in `ctest_log_parser` (external vendored in Spack) are hard-coded and that would take significant work. So generating the logs in english as you do here is likely the most efficient approach. That being said, I'd like to be able to enable this with an option, since otherwise all the build logs will be generated in English; a user might prefer to sift through logs in their native language vs. having Spack pinpoint an error in a language they are less familiar with.

Reply:
> * I'm wondering if we should explicitly set the language variables vs. unsetting them. Based on https://unix.stackexchange.com/questions/87745/what-does-lc-all-c-do, setting `LC_ALL` would override these.

Good catch. I somehow assumed that `LC_*` would override `LC_ALL` while it is the other way around. We now simply set `LC_ALL`, which should be enough.

> * Ideally the error parsing would handle other languages, but the regexes in `ctest_log_parser` (external vendored in Spack) are hard-coded and that would take significant work. So generating the logs in english as you do here is likely the most efficient approach. That being said, I'd like to be able to enable this with an option, since otherwise all the build logs will be generated in English; a user might prefer to sift through logs in their native language vs. having Spack pinpoint an error in a language they are less familiar with.

I have introduced a new config variable `build_language` that defaults to `C`. Is this what you had in mind?

Reply:
> I have introduced a new config variable build_language that defaults to C. Is this what you had in mind?

I actually had no idea how to easily address it but I think your solution was good!